labs/a7.md

```diff
@@ -0,0 +1,334 @@
+otherwise you will get a permission denied error because `root` owns the
+directory mounted over NFS)
+
+If you'd like to disconnect again, make sure you are *not* in the directory
+that has the file (otherwise it is unable to disconnect because it is still
+loaded and you will get an error message like `umount.nfs4:
+/opt/lab7/read-only: device is busy`). Then use `umount` to disconnect from
+NFS. If you run `df`, you should see that the entry that was present before has
+now disappeared.
+
+Next, mount the directory at `staff:/opt/lab7/private/<your username>` to
+`/opt/lab7/read-write` using `mount` in a similar way to before. What do you
+see in `/opt/lab7/read-write` now? Follow the instructions in the file given
+there, note that you will have to use `sudo` here too to create a new file
+since the directory mounted over NFS is owned by root, not your user.
+
+Again, if NFS takes an excessive time to mount during any of this or you cannot
+read files because it hands while doing so, please let us know on Piazza or by
+email (or at office hours if you'd prefer). We've had some problems in the past
+with NFS being very slow to mount/read and needing a restart.
+
+
+## DNS
+
+In this section we are going to be setting up our own DNS server! Remember that
+DNS is the system that maps from a domain like `ocf.berkeley.edu` to an IP like
+`169.229.226.23` (and `2607:f140:8801::1:23` for IPv6) so that computers know
+how to send information over the network to servers without people having to
+remember a bunch of numbers to connnect to everything. A more thorough
+description of this is in [Lab 5][lab-a5] if you'd like a refresher or want
+more information.
+
+[lab-a5]: /labs/a5#dns-configuration
+
+First, install the `bind9` package on your VM to set up a DNS server. By
+default, the service is not running yet. What is the `systemctl` command to
+show if the `bind9` service is running or not?
+
+In the output of the `systemctl` command, you should see that the `bind9`
+service is not running (yet) and has a unit file at
+`/lib/systemd/system/bind9.service`. If you print that file, you should see
+something like this:
+
+```
+[Unit]
+Description=BIND Domain Name Server
+Documentation=man:named(8)
+After=network.target
+Wants=nss-lookup.target
+Before=nss-lookup.target
+
+[Service]
+EnvironmentFile=/etc/default/bind9
+ExecStart=/usr/sbin/named -f $OPTIONS
+ExecReload=/usr/sbin/rndc reload
+ExecStop=/usr/sbin/rndc stop
+
+[Install]
+WantedBy=multi-user.target
+```
+
+This should look pretty familiar to you after the lecture on services! Don't
+worry if it doesn't all look familiar since there are some options you haven't
+seen yet in here, but you should at least recognize some of the options used.
+
+If you now run `dig ocf.berkeley.edu @localhost` from your VM, you should see
+that the command eventually times out after trying to run for about 15 seconds.
+The previous command failed because the DNS server was not started yet (Doh!).
```

Comment:
I'd say something like:

>The previous command failed because we specified our machine as the name server,
and the DNS server was not started yet.

A good question to ask might be why this same command succeeds when we don't use @localhost (sorry for the ping mate).

Reply:
Ah yeah, good idea on that, I'll add that to the lab checkoff form and to the lab!